app/src/main/res/layout/bottom_sheet_account.xml

```diff
@@ -0,0 +1,74 @@
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+<com.infomaniak.mail.views.BottomSheetScaffoldingView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    app:centerHorizontally="true"
+    tools:context=".ui.bottomSheetDialogs.AccountBottomSheetDialog">
+
+    <androidx.core.widget.NestedScrollView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="vertical">
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:id="@+id/recyclerViewAccount"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:clipToPadding="false"
+                android:overScrollMode="never"
+                android:paddingTop="@dimen/marginStandardMedium"
+                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                tools:itemCount="4"
+                tools:listitem="@layout/item_switch_user_account" />
+
+            <com.google.android.material.divider.MaterialDivider
+                style="@style/ItemDivider"
+                android:layout_marginVertical="@dimen/marginStandardSmall" />
+
+            <com.infomaniak.mail.ui.main.thread.actions.ActionItemView
+                android:id="@+id/addAccount"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                app:icon="@drawable/ic_add_thin"
+                app:iconPaddingEnd="@dimen/marginStandardMedium"
+                app:iconPaddingStart="@dimen/marginStandard"
+                app:text="@string/buttonAddAccount"
+                app:visibleDivider="false" />
+
+            <com.infomaniak.mail.ui.main.thread.actions.ActionItemView
+                android:id="@+id/logout"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                app:icon="@drawable/ic_logout"
+                app:iconColor="@color/redDestructiveAction"
+                app:iconPaddingEnd="@dimen/marginStandardMedium"
+                app:iconPaddingStart="@dimen/marginStandard"
+                app:text="@string/buttonAccountLogOut"
+                app:textColor="@color/redDestructiveAction"
+                app:visibleDivider="false" />
+
+        </LinearLayout>
+    </androidx.core.widget.NestedScrollView>
```

Comment:
Having the recyclerview inside the a NestedScrollView prevents the recyclerview from recycling items and only loading the visible items

Reply:
Yes we know we lack optimization here but we assume we'll not have a lot of accounts here. Even though we'll have 30/50 accounts (which we'll almost never happen), it'll work perfectly. If we want to optimize this, we have to create  an adapter taking not only accounts but also the divider and the 2 buttons. Which makes the code even more complicated for a very very small optimization. It's not worth it.